Triangulation/include/CGAL/Triangulation.h

```diff
@@ -1408,12 +1408,12 @@ operator<<(std::ostream & os, const Triangulation<TT, TDS> & tr)
 
     // infinite vertex has index 0 (among all the vertices)
     index_of_vertex[tr.infinite_vertex()] = i++;
-    os << *tr.infinite_vertex();
+    os << *tr.infinite_vertex() <<"\n";
     for( Vertex_iterator it = tr.vertices_begin(); it != tr.vertices_end(); ++it )
     {
         if( tr.is_infinite(it) )
             continue;
-        os << *it; // write the vertex
+        os << *it <<"\n"; // write the vertex
```

Comment:
You should test `is_ascii(os)` before inserting the `\n`. Because you do not want to insert end-of-lines in a binary stream.

Reply:
> Point_d I/O currently always behaves as if in ascii mode. Maybe those operators should be changed to handle binary mode if you care about that...

Sounds like a bug that should be fixed. That `Triangulation::operator<<` already has tests with `is_ascii(os)`, a few line above. I thought that was tested and supported.

Well, I am not surprised. I am the one who fixed the I/O of Triangulation_3  to support binary streams (in order to fix the I/O of Mesh_3's C3t3).